File: src/Entity/Browser/Container.php
<?php

declare(strict_types=1);

namespace Yggverse\Yoda\Entity\Browser;

use \GtkBox;
use \GtkOrientation;

use \Yggverse\Yoda\Entity\Browser;

class Container
{
    // GTK
    public GtkBox $gtk;

    // Dependencies
    public Browser $browser;

    // Requirements
    public Container\Tab $tab;

    public function __construct(
        Browser $browser
    ) {
        // Init dependency
        $this->browser = $browser;

        // Init container
        $this->gtk = new GtkBox(
            GtkOrientation::VERTICAL
        );

        // Init tab
        $this->tab = new Container\Tab(
            $this
        );

        $this->gtk->pack_start(
            $this->tab->gtk,
            true,
            true,
            0
        );

        // Render
        $this->gtk->show();
    }

    public function refresh()
    {
        // @TODO
    }
}

File: src/app/browser/container.cpp
#include "container.h"

namespace app
{
    namespace browser
    {
        // Construct
        Container::Container(
            Browser *browser
        ) {
            // Init GTK
            this->gtk = gtk_box_new(
                GTK_ORIENTATION_VERTICAL,
                Container::SPACING
            );

            // Init tabs component
            this->tab = new container::Tab(
                this
            );

            gtk_box_append(
                GTK_BOX(
                    this->gtk
                ),
                GTK_WIDGET(
                    this->tab->gtk
                )
            );

            // @TODO append testing tab
            this->tab->append(
                NULL,
                false,
                true
            );

            // Render
            gtk_widget_show(
                GTK_WIDGET(
                    this->gtk
                )
            );
        }
    }
}
